### PR TITLE
Migrate to pyproject.toml

### PR DESCRIPTION
**Pull request by gimppa**
### Description

Migrate from setup.py to pyproject.toml

### Related issues

Implements part of #765

### Type of change

- [x] New feature (non-breaking change which adds functionality)

### Changes Made

Replace setup.py and every reference to setup.py with pyproject.toml

### Testing

- [x] Tests do not apply

### Mentions

@blankdots

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+dynamic = ["version", "dependencies"]
+description = "metadata_backend"
+readme = "README.md"
+requires-python = ">=3.11"
+license = {text = "MIT License"}
+authors = [
+    { name = "CSC Developers" },
+]
+classifiers = [
+    "License :: OSI Approved :: MIT License",
```
Comment (blankdots) [inline pyproject.toml:16]:
```suggestion
    "Development Status :: 4 - Beta",
    "Intended Audience :: Developers",
    "Intended Audience :: Information Technology",
    "License :: OSI Approved :: MIT License",
    "Programming Language :: Python :: 3.11",
    "Topic :: Internet :: WWW/HTTP :: HTTP Servers",
```

a few more things here

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    "pytest==7.4.0",
+    "pytest-cov==4.1.0",
+    "pytest-xdist==3.3.1",
+    "tox==4.6.4",
+]
+docs = [
+    "sphinx >= 1.4",
+    "sphinx_rtd_theme==1.2.2",
+]
+
```
Comment (blankdots) [inline pyproject.toml:31]: i assume we can also add ?
```
[tool.black]
line-length = 120
target-version = ['py311']

[tool.isort]
atomic = true
profile = "black"
line_length = 120
py_version=311
```

**Review by blankdots — COMMENTED** [review]
should we delete setup.py ?

File: pyproject.toml [hunk 3]
```diff
@@ -34,6 +34,16 @@ docs = [
...
 
+[tool.black]
+line-length = 120
+target-version = ['py311']
+
+[tool.isort]
+atomic = true
+profile = "black"
+line_length = 120
+py_version=311
```
Comment (gimppa) [inline pyproject.toml:45]: So this [broke workflows](https://github.com/CSCfi/metadata-submitter/actions/runs/5642935238/job/15283779500?pr=781#step:7:18) and I have no idea why.

Comment (blankdots) [reply pyproject.toml:45]: it checks the test files from what i see:
so either we exclude them: https://github.com/CSCfi/metadata-submitter/actions/runs/5642935238/job/15283779500?pr=781#step:7:18 or some other fix

**Pull request merged**